python/AmcCarrierCore/AppTop/TopLevel.py

```diff
@@ -56,99 +59,16 @@ def __init__(   self,
         self._numRxLanes = numRxLanes
         self._numTxLanes = numTxLanes
         self._numWaveformBuffers = numWaveformBuffers
-        
-        rssiInterlaved    = False
-        rssiNotInterlaved = False
-        
-        # Check for valid link range
```

Comment:
@slacrherbst: Does this mean we potential have to copy this code in every repo instead of sourcing from a common submodule location?

Reply:
I kind of feel like there should be a way to distrubute a "common" root class to different application git repos

Reply:
yes this is bugging me as well. I am trying to think of a clean solution. The problem with putting the link code in the devices is it makes it difficult to use the monitor Device wrappers without confusing the tree. I might end up putting Root base classes in the AmcCarrierCore package. This would open up the option of distributing base Root classes in the firmware dist that are then subclassed. in the application code and connected as required to data writers, etc.